### PR TITLE
Refactor/is palette unique function

### PR DESCRIPTION
**Pull request by SethBurns**
## What is the feature?
Refactor isPaletteUnique function to call areArraysEqual function to eliminate double for loop.
Changed all `for (i = 0` to `for (var i = 0` to prevent `i` from being a global variable, or to scope `i` to each function individually.

## How you implemented the solution?
Separated out the individual for loops inside isPaletteUnique function.

## Does it impact any other area of the project?
It should only prevent future bugs where `i` was declared as a global variable.

## Any changes in the UI (Screenshots)?
None
## How to test the feature (A test scenario or any new setup)?
Save palette, save palette, click on saved palette, save palette again. It shouldn't save.

### CODE REVIEW

**Review by mbenfowler — APPROVED** [review]
Ahh yeah this is great!!  Approved pending removal of console logs

File: main.js [hunk 1]
```diff
@@ -113,18 +113,22 @@ function savePalette() {
...
-        var matches = true;
-        for (j = 0; j < palettesList[i].length; j++) {
-            if (palettesList[i][j] !== singlePalette[j]) {
-                matches = false;
-                break;
-            }
-        }
-        if (matches) {
+    for (var i = 0; i < palettesList.length; i++) {
+        console.log('isPaletteUnique', i);
```
Comment (jalbe0076) [inline main.js]: We need to remove the `console.log`

File: main.js [hunk 2]
```diff
@@ -113,18 +113,22 @@ function savePalette() {
...
+        if (areArraysEquivalent(palettesList[i], singlePalette)) {
+            return false;
+        } 
+    }
+    return true;   
+}    
+
+function areArraysEquivalent(array1, array2) {
+    for (var i = 0; i < array1.length; i++) {
+        console.log('areArraysEquivalent', i);
```
Comment (jalbe0076) [inline main.js]: We need to remove the `console.log`

File: main.js [hunk 3]
```diff
@@ -113,18 +113,22 @@ function savePalette() {
...
-        for (j = 0; j < palettesList[i].length; j++) {
-            if (palettesList[i][j] !== singlePalette[j]) {
-                matches = false;
-                break;
-            }
-        }
-        if (matches) {
+    for (var i = 0; i < palettesList.length; i++) {
+        console.log('isPaletteUnique', i);
+        if (areArraysEquivalent(palettesList[i], singlePalette)) {
```
Comment (jalbe0076) [inline main.js:119]: image I did not know we could invoke a function inside a conditional to get a true or false. Amazing!!!

File: main.js [hunk 4]
```diff
@@ -113,18 +113,22 @@ function savePalette() {
...
+            return false;
+        } 
+    }
+    return true;   
+}    
+
+function areArraysEquivalent(array1, array2) {
```
Comment (jalbe0076) [inline main.js]: we should make the parameters more descriptive. Maybe array1 = paletteToCheck & array2 = currentPalette?

**Review by jalbe0076 — CHANGES_REQUESTED** [review]

File: main.js (hunk 3 above)
Comment (mbenfowler) [reply main.js:119]: me neither woo!

**Pull request merged**